public/app/features/alerting/unified/components/contact-points/useContactPoints.tsx

```diff
@@ -3,21 +3,87 @@
  * and (if available) it will also fetch the status from the Grafana Managed status endpoint
  */
 
-import { NotifierType, NotifierStatus } from 'app/types';
+import { produce } from 'immer';
+import { remove } from 'lodash';
 
-// A Contact Point has 1 or more integrations
-// each integration can have additional metadata assigned to it
-export interface ContactPoint<T extends Notifier> {
-  notifiers: T[];
-}
+import { alertmanagerApi } from '../../api/alertmanagerApi';
+import { GRAFANA_RULES_SOURCE_NAME } from '../../utils/datasource';
 
-interface Notifier {
-  type: NotifierType;
-}
+import { enhanceContactPointsWithStatus } from './utils';
+
+export const RECEIVER_STATUS_KEY = Symbol('receiver_status');
+const RECEIVER_STATUS_POLLING_INTERVAL = 10 * 1000; // 10 seconds
+
+/**
+ * This hook will combine data from two endpoints;
+ * 1. the alertmanager config endpoint where the definition of the receivers are
+ * 2. (if available) the alertmanager receiver status endpoint, currently Grafana Managed only
+ */
+export function useContactPointsWithStatus(selectedAlertmanager: string) {
+  const isGrafanaManagedAlertmanager = selectedAlertmanager === GRAFANA_RULES_SOURCE_NAME;
+
+  // fetch receiver status if we're dealing with a Grafana Managed Alertmanager
+  const fetchContactPointsStatus = alertmanagerApi.endpoints.getContactPointsStatus.useQuery(undefined, {
```

Comment:
I actually prefer using the `.endoints.<function>.useQyuery()` because it allows me to jump directly to the function definition – my editor can't seem to resolve `useGetContactPointsStatus` :(

Reply:
I'm afraid this is a drawback of RTK query generated hooks. They are generated dynamically by RTK query, hence navigation doesn't work